lib/rack/jsonp.rb

```diff
@@ -60,7 +62,7 @@ def pad(callback, response, body = "")
         body << s.to_s.gsub("\u2028", '\u2028').gsub("\u2029", '\u2029')
       end
       close(response)
-      ["#{callback}(#{body})"]
+      ["/**/#{callback}(#{body})"]
```

Comment:
I'm not sure if I feel that this is incredibly clever, or to feel disgusted by browser security.

Reply:
Wait untill you learn why non jsonp xhr response should all purposely crash the JS interpreter. Facebook prepends an infinite loop `for(;;;); `